// ngx-qrcode-reader/esm2015/lib/ngx-qrcode-reader.component.js
import { Component, Input, Output, EventEmitter, ViewChild, Renderer2 } from '@angular/core';
var QrCode = require('qrcode-reader').default;
export class NgxQrcodeReaderComponent {
    constructor(renderer) {
        this.renderer = renderer;
        this.cssClass = 'qrcode-reader';
        this.showQRCode = true;
        this.fileValue = '';
        this.qrContentResult = new EventEmitter();
        this.qrReader = new QrCode();
    }
    ngOnInit() {
        let that = this;
        this.qrReader.callback = (error, result) => {
            if (error) {
                that.qrContentResult.emit({ error: error });
                console.error(error);
                return;
            }
            that.qrContentResult.emit({ content: result.result, type: "text" });
        };
    }
    ngOnChanges() {
        this.decode(this.fileValue);
        if (this.showQRCode) {
            this.render();
        }
    }
    decode(value) {
        this.qrReader.decode(value);
    }
    render() {
        let element;
        element = this.renderer.createElement('img');
        element.setAttribute("src", this.fileValue);
        this.renderElement(element);
    }
    renderElement(element) {
        for (let node of this.qrCodeElement.nativeElement.childNodes) {
            this.renderer.removeChild(this.qrCodeElement.nativeElement, node);
        }
        this.renderer.appendChild(this.qrCodeElement.nativeElement, element);
    }
}
NgxQrcodeReaderComponent.decorators = [
    { type: Component, args: [{
                selector: 'ngx-qrcode-reader',
                template: `<div #qrCodeElement [class]="cssClass"></div>`
            },] }
];
NgxQrcodeReaderComponent.ctorParameters = () => [
    { type: Renderer2 }
];
NgxQrcodeReaderComponent.propDecorators = {
    cssClass: [{ type: Input, args: ['qrClass',] }],
    showQRCode: [{ type: Input, args: ["showQr",] }],
    fileValue: [{ type: Input, args: ['qrFile',] }],
    qrCodeElement: [{ type: ViewChild, args: ['qrCodeElement', { static: true },] }],
    qrContentResult: [{ type: Output, args: ['onfetchQRContent',] }]
};
